lib/Core/SQLiteBuildDB.cpp

```diff
@@ -338,18 +386,16 @@ class SQLiteBuildDB : public BuildDB {
         abort();
       }
       assert(sqlite3_column_count(findRuleDependenciesStmt) == 1);
-      result_out->dependencies.push_back(
-          std::string(
-              (const char*)sqlite3_column_text(findRuleDependenciesStmt, 0),
-              sqlite3_column_bytes(findRuleDependenciesStmt, 0)));
+      uint64_t keyID = sqlite3_column_int64(findRuleDependenciesStmt, 0);
```

Comment:
I would call this `dependencyID` just so it doesn't shadow the very similar but semantically different `keyID` in the containing scope.